cmd/config-versions.go

```diff
@@ -927,3 +924,12 @@ type serverConfigV33 struct {
 		// Add new external policy enforcements here.
 	} `json:"policy"`
 }
+
+type openIDConfig struct {
+	// JWKS validator config.
+	JWKS         validator.JWKSArgs `json:"jwks"`
+	ClientID     string             `json:"clientID"`
```

Comment:
We cannot save this in our config @kanagarajkm both clientID and clientSecret , this should be provided by the user on the UI 

Here there are two types of authentication one is

- WebIdentity
- ClienGrants

WebIdentity requires that you show an identity login page as soon as they reach our browser
ClientGrants, on the other hand, doesn't require this behavior, because client grants is a static access/secret style combination but provided as way to be revocable remotely. 

So this config change is not needed, you only need to take one AuthURL and TokenURL. This can be obtained by taking in OpenID configuration URL https://github.com/minio/minio/issues/7055

This would allow you to parse it once @kanagarajkm

Reply:
> We cannot save this in our config @kanagarajkm both clientID and clientSecret , this should be provided by the user on the UI
> 
> Here there are two types of authentication one is
> 
> * WebIdentity
> * ClienGrants
> 
> WebIdentity requires that you show an identity login page as soon as they reach our browser
>
> ClientGrants, on the other hand, doesn't require this behavior, because client grants is a static access/secret style combination but provided as way to be revocable remotely.

@harshavardhana I feel asking the user to enter Client ID and Client Secret every time they login through Identity Provider is not a seamless experience. Also I am not sure if a regular user will have access to Client Secret.
 
I thought we are going support only WebIdentity through browser not Client grants. Client grants are more of a machine to machine (or client to application) auth mechanism where each client gets a set of credentials from the Identity provider and there is no direct human interaction.

I would like to understand your thoughts on this, thanks.

> So this config change is not needed, you only need to take one AuthURL and TokenURL. This can be obtained by taking in OpenID configuration URL #7055
> 
> This would allow you to parse it once @kanagarajkm

yes, i will do that, thanks!

Reply:
> @harshavardhana I feel asking the user to enter Client ID and Client Secret every time they login through Identity Provider is not a seamless experience. Also I am not sure if a regular user will have access to Client Secret.

client id and client secrets are not meant for regular user logins. They are only meant for applications and applications won't be using our Web UI. 

Web UI we should limit to WebIdentity, LDAP/AD - if you want to extend this to client id and client secret them you have to ask that instead of access/secret.

Also client ID and client secret do not involve a callback mechanism for OAuth, client grants provide tokens directly to the caller without a callback RBAC exchange.

in this PR I would simply expect that WebIdentity is supported.